gmi100.c

```diff
@@ -7,7 +7,7 @@
 #define WARN(msg) do { fputs("WARNING: "msg"\n", stderr); goto start; } while(0)
 
 int main(int argc, char **argv) {
-        char uri[1024+1], buf[1024+1], buf2[1024+1], *bp, *t=tmpnam(0);
+        char uri[1024+1], buf[2*(1024+1)], buf2[1024+1], *bp, *t=tmpnam(0);
```

Comment:
Yep, this works.
I think that we could just use bigger buffers.
The 4096 is a default for many internal POSIX functions, at least in glibc.

The `+1` in `uri` is only to clearly show that URI max length is 1024 according to Gemini specification `+1` for null terminator.

```suggestion
        char uri[1024+1], buf[4096], buf2[2048], *bp, *t=tmpnam(0);
```